hal/ia32/cpu.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * CPU related routines
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <syspage.h>
+
+
+/* FIXME: temporary usage until new syspage implementation */
+extern int hal_memoryGetEntry(unsigned int *offs, mem_entry_t *entry);
+
+
+__attribute__((noreturn)) void hal_cpuJump(addr_t addr)
+{
+	/* FIXME: temporary HAL syspage usage, should be removed after new syspage implementation is added */
+	void *syspage = (void *)syspage_getAddress();
+	unsigned short *mmsz = syspage + 300;
```

Comment:
void pointer arithmetic. Cast to char * for arithmetic. Also, where this 300 came from? Replace this magic number with something of significance

Reply:
I changed the function a bit - it's cleaner now. Please note that all the code except inline asm with storing syspage address on stack and final jump is going to be removed once new syspage implementation is added - it's there for compatibility with current kernel syspage implementation.